hal/armv7a/imx6ull/imx6ull.c

```diff
@@ -0,0 +1,363 @@
+	*(imx6ull_common.iomux + pad_dram_odt0) = 0x00000030;
+	*(imx6ull_common.iomux + pad_dram_odt1) = 0x00000030;
+
+	*(imx6ull_common.iomux + grp_ctlds) = 0x00000030;
+	*(imx6ull_common.iomux + grp_ddrmode_ctl) = 0x00020000;
+
+	*(imx6ull_common.iomux + pad_dram_sdqs0p) = 0x00000030;
+	*(imx6ull_common.iomux + pad_dram_sdqs1p) = 0x00000030;
+
+	*(imx6ull_common.iomux + grp_ddrmode) = 0x00020000;
+	*(imx6ull_common.iomux + grp_b0ds) = 0x00000030;
+	*(imx6ull_common.iomux + grp_b1ds) = 0x00000030;
+
+	*(imx6ull_common.iomux + pad_dram_dqm0) = 0x00000030;
+	*(imx6ull_common.iomux + pad_dram_dqm1) = 0x00000030;
+
+	/* Config DDR control register */
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x00008000;
+	*(imx6ull_common.mmdc + mmdc_mpzqhwctrl) = 0xa1390003;
+	*(imx6ull_common.mmdc + mmdc_mpwldectrl0) = 0x00150019;
+	*(imx6ull_common.mmdc + mmdc_mpdgctrl0) = 0x41550153;
+	*(imx6ull_common.mmdc + mmdc_mprddlctl) = 0x40403a3e;
+	*(imx6ull_common.mmdc + mmdc_mpwrdlctl) = 0x40402f2a;
+
+	*(imx6ull_common.mmdc + mmdc_mprddqby0dl) = 0x33333333;
+	*(imx6ull_common.mmdc + mmdc_mprddqby1dl) = 0x33333333;
+
+	*(imx6ull_common.mmdc + mmdc_mpwrdqby0dl) = 0xf3333333;
+	*(imx6ull_common.mmdc + mmdc_mpwrdqby1dl) = 0xf3333333;
+
+	*(imx6ull_common.mmdc + mmdc_mpdccr) = 0x00944009;
+	*(imx6ull_common.mmdc + mmdc_mpmur0) = 0x00000800;
+
+	*(imx6ull_common.mmdc + mmdc_mdpdc) = 0x0002002d;
+	*(imx6ull_common.mmdc + mmdc_mdotc) = 0x1b333030;
+	*(imx6ull_common.mmdc + mmdc_mdcfg0) = 0x676b52f3;
+	*(imx6ull_common.mmdc + mmdc_mdcfg1) = 0xb66d0b63;
+	*(imx6ull_common.mmdc + mmdc_mdcfg2) = 0x01ff00db;
+	*(imx6ull_common.mmdc + mmdc_mdmisc) = 0x00201740;
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x00008000;
+	*(imx6ull_common.mmdc + mmdc_mdrwd) = 0x000026d2;
+	*(imx6ull_common.mmdc + mmdc_mdor) = 0x006b1023;
+	*(imx6ull_common.mmdc + mmdc_mdasp) = 0x00000047;
+
+	*(imx6ull_common.mmdc + mmdc_mdctl) = 0x82180000;
+	*(imx6ull_common.mmdc + mmdc_mppdcmpr2) = 0x00400000;
+
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x2008032;
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x8033;
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x48031;
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x15208030;
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x4008040;
+
+	*(imx6ull_common.mmdc + mmdc_mdref) = 0x800;
+
+	*(imx6ull_common.mmdc + mmdc_mpodtctrl) = 0x227;
+	*(imx6ull_common.mmdc + mmdc_mdpdc) = 0x2552d;
+	*(imx6ull_common.mmdc + mmdc_mapsr) = 0x11006;
+	*(imx6ull_common.mmdc + mmdc_mdscr) = 0x0;
+}
+
+
+static void imx6ull_pllInit(void)
+{
+/* This code bases on initialization PLLs and clocks from phoenix-rtos-kernel (imx6ull/_init.S).
+ * It occurred that it doesn't work as expected. PLL should be locked after initialization but it doesn't.
+ * PLO will used clock and PLL configuration set by bootROM.
+ * NOTE: TODO - PLL and clocks configuration should be checked! */
+#if 0
+	/* Set ARM clock to 792 MHz; set ARM clock divider to 1 */
+	*(imx6ull_common.ccm + ccm_cacrr) = 0;
+	hal_cpuDataSyncBarrier();
```

Comment:
I think we don't need to disable this one

Reply:
Done